src/InstallWizard.js

```diff
@@ -329,13 +329,21 @@ class InstallWizard extends Component {
   // Pages within the installer may request that the model be forceably loaded
   // from disk, espcially when a change is made to directly to the model files
   // to the model.  Returns a promise
-  loadModel = () => fetchJson('/api/v2/model')
-    .then(responseData => {
-      this.setState({'model': fromJS(responseData)});
-    })
-    .catch((error) => {
-      console.log('Unable to retrieve saved model');// eslint-disable-line no-console
-    })
+  loadModel() {
+    // If there is a pending promise to fetch the model return that promise.
+    if(this.fetchModelPromise?.done) {
```

Comment:
Shouldn't this be `! this.fetchModelPromise?.done` ? This check would seem to cause the code to be executed even if it has already been resolved.

Reply:
No. When this method is called the caller wants the model to be loaded, so we will load it. If there is a existing promise that has not finished we will not create another request to load the model. We are already loading the model.

Reply:
That being said there may have been a logic error in there, so I have made that a little more explicit.